spec/lib/whitehall_importer/integrity_checker_spec.rb

```diff
@@ -17,19 +17,19 @@
 
   describe "#valid?" do
     let(:edition) do
-      build(
-        :edition,
-        document_type: document_type,
-        tags: {
-          primary_publishing_organisation: [SecureRandom.uuid],
-          organisations: [SecureRandom.uuid],
-        },
-      )
+      build(:edition,
+            document_type: document_type,
+            document: create(:document, first_published_at: "2020-03-11 12:00:00 +0000"),
```

Comment:
I notice in the commit message it says: "first_published_at time is returned from publishing-api in the format -"2020-03-11 12:00:00 +0000" and so the specs are using this same format for the pub-api edition." Which I wonder if this is why you've set these to these strings?

I'm a bit baffled by this though as these times are used to create Time objects in Content Publisher DB and have no relation with Publishing API and the Publishing API first_published_at time is formatted in ISO8601: https://github.com/alphagov/publishing-api/blob/6e34eb8c2593a3bfc38a28fc11de0989e927adcd/app/presenters/queries/content_item_presenter.rb#L119

Reply:
I've set the times in publishing_api_item to be consistent with the times that are returned from the actual publishing_api which I think is the best thing to do as it resembles real life. 

I know this value sets a time object in ruby when creating a document but I was trying to be consistent by only using strings to set timestamps in this spec and the format - "2020-03-11 12:00:00 +0000" is what's returned when you call to_s on a Time object. The second reason I used a string is you can't use one of ruby / rails neat time helpers like `Time.zone.now.at_noon` as you need to pass a specific date as the one you are comparing against is a string in publishing_api_item. You could pass Time a specific date but it just seemed to be easier to just write this as a string, what do you think?

Reply:
Ok, I don't think you've quite captured what part of this I've found confusing. It's mostly where the commit message seems to differ from what actually happens / what we use and me trying to piece together whether you actually mean that or not.

The commit message says:

> "first_published_at time is returned from publishing-api in the format -"2020-03-11 12:00:00 +0000"

But we don't use first_published_at from the Publishing API, nor is it in that format. We seem to care about first_public_at which is in this format: "2020-01-24T16:38:00.000+00:00"

Passing strings seems fine, though you don't need quite that degree of granularity - "2020-03-11 12:00:00 +0000" produces the same output as "2020-03-11 12:00" since the time is in GMT. If you wanted it to be safe for all parts of the year you could do say: `Time.utc(2020, 3, 11).noon` or "2020-03-11 12:00 UTC"

Reply:
Oh I see what you mean, I've updated the commit message accordingly, nice catch! 
I've also updated the times to use "2020-03-11 12:00 UTC" format.